x/dispensation/types/records.go

```diff
@@ -87,3 +102,26 @@ func IsValidDistribution(distributionType string) (DistributionType, bool) {
 		return DistributionType_DISTRIBUTION_TYPE_UNSPECIFIED, false
 	}
 }
+func IsValidDistribution(distributionType string) (DistributionType, bool) {
+	switch distributionType {
+	case "DISTRIBUTION_TYPE_AIRDROP":
+		return DistributionType_DISTRIBUTION_TYPE_AIRDROP, true
+	case "DISTRIBUTION_TYPE_LIQUIDITY_MINING":
+		return DistributionType_DISTRIBUTION_TYPE_LIQUIDITY_MINING, true
+	case "DISTRIBUTION_TYPE_VALIDATOR_SUBSIDY":
+		return DistributionType_DISTRIBUTION_TYPE_VALIDATOR_SUBSIDY, true
+	default:
+		return DistributionType_DISTRIBUTION_TYPE_UNSPECIFIED, false
+	}
+}
+
+func IsValidClaim(claimType string) (DistributionType, bool) {
```

Comment:
Should be called IsValidClaimType() ... or even better, converted to enum then use claimType.Valid()

Reply:
I changed the name , not sure if I understand the second option 
Claim and Distribution both use the same type , and Airdrop is not Valid for distribution.  The Valid function would need a parameter in that case .